### ISSUE TITLE
[Bug] Widescreen hack does not work with DX11/GL

### ISSUE THREAD

**Issue by diego-au** [libretro/dolphin#282]
Enabling "Widescreen Hack" causes a stretched image rather than an extension of the camera / viewport when using DX11 or GL. This issue is not present when using the Vulkan driver.

As Vulkan is not supported on all platforms such as Xbox Series X|S, it would be good to resolve

This issue is very noticeable with Gamecube games:

Widescreen Hack with Vulkan Driver: Notice the hole/opening in the wall on the left is visible 
![image](https://user-images.githubusercontent.com/112910797/188522065-c31d0e6c-f772-4148-abaa-47c9e2ce06f7.png)

Widescreen Hack with DX11: Opening in the left wall is not visible as image is stretched 
![image](https://user-images.githubusercontent.com/112910797/188522102-3c2e1c9d-c393-4650-b567-1606df5e9583.png)

**Comment by bslenul** [comment]
Yeah I mentioned the issue in #239 (although I should've make 2 separate issues instead of a 2-in-1, for visibility purpose), I tried fixing it myself multiple times but this is too complicated for my very, very limited skills :p I hope someone will be able to fix this at some point...
